web-marketplace/src/pages/ProfileEdit/ProfileEdit.Main.tsx

```diff
@@ -148,15 +149,20 @@ export const ProfileEditMain = () => {
     fileNamesToDeleteRef,
   });
 
+  const onUpdate = (isFormUpdated: boolean) => {
+    setIsSaveDisabled(!isFormUpdated);
+  };
+
   return (
     <EditProfileForm
       onSubmit={onSubmit}
       onSuccess={onSuccess}
       onUpload={onUpload}
+      onUpdate={onUpdate}
       initialValues={initialValues}
       isDirtyRef={isDirtyRef}
     >
-      <EditProfileFormActionBar />
+      <EditProfileFormActionBar saveDisabled={isSaveDisabled} />
```

Comment:
I think you could just use `isDirtyRef` to check whether to disable the button or not